runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -516,6 +525,16 @@ static StreamingDataflowWorker forTesting(
             options.getWindmillHarnessUpdateReportingPeriod().getMillis(),
             options.getPerWorkerMetricsUpdateReportingPeriodMillis());
 
+    GrpcWindmillStreamFactory windmillStreamFactory =
+        createWindmillStreamFactoryBuilder(options, 1)
+            .setProcessHeartbeatResponses(
+                new WorkHeartbeatResponseProcessor(computationStateCache::get))
+            .build();
+    if (options.isEnableStreamingEngine()) {
+      windmillStreamFactory.scheduleHealthChecks(
```

Comment:
This looks like is missing on non test flow.

Reply:
fixed